src/Linters/PackagePrivateLinter.hack

```diff
@@ -0,0 +1,318 @@
+        $parts = vec[];
+
+        foreach ($use_decl->getChildrenOfType(QualifiedName::class) as $qualified_name) {
+          foreach ($qualified_name->getDescendantsOfType(NameToken::class) as $name_token) {
+            $parts[] = $name_token->getText();
+          }
+        }
+
+        foreach ($use_decl->getDescendantsOfType(NamespaceUseClause::class) as $use_clause) {
+          $parts_item = vec[];
+          $name = $use_clause->getName();
+          if ($name is NameToken) {
+            $parts_item[] = $name->getText();
+          } else if ($name is QualifiedName) {
+            foreach ($name->getDescendantsOfType(NameToken::class) as $name_token) {
+              $parts_item[] = $name_token->getText();
+            }
+          }
+
+          $parts_item = Vec\concat($parts, $parts_item);
+          if (C\count($parts_item) > 1) {
+            $alias = $use_clause->getAlias();
+            $kind = $use_kind($use_decl);
+            if ($kind is nonnull) {
+              $shape_obj = shape(
+                'parts' => $parts_item,
+                'alias' => $alias?->getText(),
+                'node' => $use_clause,
+              );
+              if ($kind === 'namespace') {
+                $key = $parts[C\count($parts) - 1];
+                $uses['namespace'][$key] = $shape_obj;
+              } else if ($kind === 'function') {
+                $uses['function'][] = $shape_obj;
+              } else {
+                $uses['type'][] = $shape_obj;
+              }
+            }
+          }
+        }
+      }
+    }
+
+    return $uses;
+  }
+
+  /**
+   * This function takes a qualified name token and flattens it into a vec of string, ignoring all '\'.
+   * eg: \Test1\Private would become a vec[Test1, Private];
+   */
+  private function getQualifiedNameFlatten(
+    QualifiedName $qualified_name,
+    dict<string, use_definition_parts_t>
+    $use_namespace_definitions_for_file,
+    vec<string> $current_namespace_for_file,
+  ): vec<string> {
+    $parts = vec[];
+    if ($qualified_name->hasParts()) {
+      $absolute_path = false;
+      $qualified_name_node = $qualified_name->getPartsx()->getChildren();
+
+      foreach ($qualified_name_node as $index => $qualified_name_parts) {
+        if ($index === 0 && $qualified_name_parts->getItem() is null
+          && $qualified_name_parts->getSeparator()?->getText() === '\\'
+        ) {
+          $absolute_path = true;
+        }
+        $item = $qualified_name_parts->getItem();
+        if ($item is nonnull) {
+          $parts[] = $item->getText();
+        }
+      }
+
+      if (!$absolute_path && C\count($parts) > 0) {
+        /**
+         * get the absolute path by checking the following:
+         * 1. Is the path mentioned in qualified name relative to one of the namespace in use namespace statement(s).
+         * 2. Is the path mentioned in qualified name relative to the namespace of the current file.
+         */
+        $path_found = false;
```

Comment:
this isn't complete, especialyl if there are multiple namespace blocks - the resolve_type/resolve_function functions mentioned above do this

Reply:
removed this function, and considered multiple namespaces in a single file.